torch/csrc/jit/batched/BatchTensor.cpp

```diff
@@ -0,0 +1,70 @@
+#include "BatchTensor.h"
+
+namespace torch { namespace jit {
+
+BatchTensor::BatchTensor(at::Tensor data, at::Tensor mask, std::vector<bool> dims){
+  if(data.dim() != mask.dim() || mask.dim() != int64_t(dims.size()) + 1){
+    throw std::runtime_error("malformed MaskedBatch with data: "
+      + std::string(data.toString()) + " mask: " + mask.toString());
+  }
+  this->data = data;
+  this->mask = mask;
+  this->dims = dims;
+}
+
+BatchTensor::BatchTensor(std::vector<at::Tensor> datalist, std::vector<bool> dims) {
```

Comment:
Take `datalist` here as const reference.

Later, since you're making an internal copy of `dims`, it's okay to take that as value and move out of this instance into the class member

```
 BatchTensor::BatchTensor(const std::vector<at::Tensor>& datalist, std::vector<bool> dims) { 
```

Reply:
As we just discussed, we need `dims` to be Tensor instead of std::vector<bool>. I'll change that to Tensor.